pkg/kubefed2/federate/federate.go

```diff
@@ -136,11 +145,11 @@ func (j *federateResource) Run(cmdOut io.Writer, config util.FedConfig) error {
 		Name:      j.resourceName,
 	}
 
-	_, err = FederateResource(hostConfig, qualifiedTypeName, qualifiedResourceName, j.DryRun)
+	_, err = FederateResource(cmdOut, hostConfig, qualifiedTypeName, qualifiedResourceName, j.outputYAML, j.DryRun)
 	return err
 }
 
-func FederateResource(hostConfig *rest.Config, qualifiedTypeName, qualifiedName ctlutil.QualifiedName, dryrun bool) (*unstructured.Unstructured, error) {
+func FederateResource(cmdOut io.Writer, hostConfig *rest.Config, qualifiedTypeName, qualifiedName ctlutil.QualifiedName, outputYAML, dryrun bool) (*unstructured.Unstructured, error) {
```

Comment:
I still think this method should be decomposed for consistency with `enable`, such that generating the fed resource is one method call and creating it is another.  Fine to have a convenience method that calls generate and then enable, but I think generating yaml output should only be in the cli-invoked Run.

Reply:
I was in the process of doing this change in `federate ns --content` already, because it works well for multiple resources. Nevertheless moved it here in the simpler form here now.